lib/Parse/ParseDecl.cpp

```diff
@@ -273,6 +279,20 @@ void Parser::parseTopLevelItems(SmallVectorImpl<ASTNode> &items) {
       consumeToken();
     }
   }
+
+#if SWIFT_SWIFT_PARSER
+  // Perform round-trip checking.
+  if (Context.LangOpts.hasFeature(Feature::ParserRoundTrip) &&
+      SF.exportedSourceFile &&
+      !L->lexingCutOffOffset() &&
```

Comment:
Do we still need this? The new parser is also guarding itself against stack overflows.